docs/partitions.md

```diff
@@ -0,0 +1,92 @@
+# Partitions
+
+In the original implementation of ringpop, if a cluster is split to multiple
+partitions, nodes in each partition declare each other as faulty, and don't
+communicate. Ringpop implemented support for merging the partitions, which we
+call `healing`.
+
+## Introdction -- basic algorithm
+
+In order for two partitions to heal, the algorithm does the following,
+periodically (number of times per time unit is upper bounded):
+
+1. Randomly select a `faulty` node.
+2. Send it a `/join` request, get its membership list.
+3. If the local and retrieved lists are incompatible (merging them will 
+   introduce new faulties), force all cluster to reincarnate by marking
```

Comment:
Mark all incompatible nodes suspect, when receiving this change the respective node will reassert that it is actually alive and update its incarnation number making it compatible for merge.